lib/tasks/adhoc.rake

```diff
@@ -0,0 +1,15 @@
+namespace :adhoc do
+  desc 'Extract a list of All articles and what categories they sit in'
+  task :extract_page => :environment do
+    File.open('pages_with_cat.csv','w') do |file|
+      file.write "Url\tCategories\n"
+      en = Comfy::Cms::Site.where(label: :en).first
+      Comfy::Cms::Page.where(site: en, state: :published).find_each do |page|
```

Comment:
I use `find_each` because it uses `find_in_batches` underneath which is suited for this type of processing.